src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/ProcessExecutor.cs

```diff
@@ -0,0 +1,206 @@
+using Microsoft.Extensions.Logging;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Text;
+
+namespace Azure.Tools.GeneratorAgent
+{
+    /// <summary>
+    /// Executes external processes with comprehensive logging, timeout support, and proper resource management.
+    /// </summary>
+    internal class ProcessExecutor
+    {
+        private readonly ILogger<ProcessExecutor> Logger;
+
+        public ProcessExecutor(ILogger<ProcessExecutor> logger)
+        {
+            Logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        /// <summary>
+        /// Executes a command asynchronously with comprehensive error handling and logging.
+        /// </summary>
+        /// <param name="command">The command to execute.</param>
+        /// <param name="arguments">The command arguments (can be null or empty).</param>
+        /// <param name="workingDirectory">The working directory for the command (uses current directory if null).</param>
+        /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
+        /// <param name="timeout">Optional timeout for the command execution.</param>
+        /// <returns>A tuple containing success status, standard output, and error output.</returns>
+        /// <exception cref="ArgumentException">Thrown when command is null or whitespace.</exception>
+        public virtual async Task<(bool Success, string Output, string Error)> ExecuteAsync(
+            string command,
+            string arguments,
+            string? workingDirectory,
+            CancellationToken cancellationToken,
+            TimeSpan? timeout = null)
+        {
+            if (string.IsNullOrWhiteSpace(command))
+                throw new ArgumentException("Command cannot be null or empty", nameof(command));
+
+            arguments ??= string.Empty;
+
+            try
+            {
+                StringBuilder outputBuilder = new StringBuilder(capacity: 4096);
+                StringBuilder errorBuilder = new StringBuilder(capacity: 1024);
+                object outputLock = new object();
+                object errorLock = new object();
```

Comment:
```suggestion
                object outputLock = new();
                object errorLock = new();
```

Since you're not a `var` fan, I'll mention this is something that you can also do to safe keystrokes.   Your call, just mentioning for awareness.